### ISSUE TITLE
Fix error for some CSS character escapes

### ISSUE THREAD

**Issue by nwmcsween** [stylelint/stylelint#4202]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Stylelint cannot parse some CSS character escapes

> Which rule, if any, is this issue related to?

None

> What CSS is needed to reproduce this issue?

```css
.w\3a 25 {
  width: 25%;
}
```

> What stylelint configuration is needed to reproduce this issue?

Any

> Which version of stylelint are you using?

Any

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Makefile stylelint-cli

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No

> What did you expect to happen?

Parse escape with spaces

> What actually happened (e.g. what warnings or errors you are getting)?

Unexpected unknown type selector "25"

**Comment by evilebottnawi** [comment]
Bug in postcss-selector-parser, need update parser

**Comment by vankop** [comment]
@hudochenkov can we label this and https://github.com/stylelint/stylelint/issues/3310 as issues that blocked by `postcss-selector-parser` update

**Comment by hudochenkov** [comment]
It's label as blocked, and your comments have the reason :)

**Comment by vankop** [comment]
I just suggested to optimize your work and update maintainer workflow in case of updating `postcss-*-parser`, for instance:

1. apply update
2. review, close or remove block status from issues dependent on this update

Plus it gives more information for contributors/community what will be fixed or could be done after update

**Comment by jeddy3** [comment]
[Fixed](https://stylelint.io/demo#N4Igxg9gJgpiBcIB0B3AOmgzAQwAQCYBWXYNAO11xQEsoAXAC3gMIFIBucgXxABoQAzgE8ydbAA8E4AQL7gIZAGbUA5lNIVcaEACcArgBsYA7cw2VK2gTCNg6EHQFo6QgA4xHZCI71kA1l4oZKa4ANrkFhZ0+jC8EZEk8QlaIKpeOjAh4ZrJluB6AvYAto42MEUwoiYgSQkAurWUXLUNms1kPFxAA).
